### PR TITLE
chore: Cut v2.11.0 

### LINKED ISSUE #2293: Timeline for a release of Kube State Metrics that uses v1.28 of the Kubernetes client-go package?
**What would you like to be added**:
Hi 👋 ,

Do you have a timeline for when a versioned release of the latest `main` branch will be made (I'm guessing `v2.11.0`) containing `v1.28` of the Kubernetes client-go package (as detailed on [this table](https://github.com/kubernetes/kube-state-metrics?tab=readme-ov-file#compatibility-matrix))

**Why is this needed**:
We wish to upgrade our EKS Cluster to v1.28 of Kubernetes and want to ensure the Client Library used by Kube State Metrics is compatible

**Describe the solution you'd like**
The latest code in main to have a versioned release we can reference in an install of this tool

**Additional context**

### PR DESCRIPTION
**Pull request by rexagod**
Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

***

Fixes #2293

### CODE REVIEW

**Comment by rexagod** [comment]
`v2.12.0+` will be released through an automated workflow, pushing shortly.

**Comment by krishnaindani** [comment]
Is there any plan to fix https://github.com/advisories/GHSA-5f94-vhjq-rpg8 and https://github.com/advisories/GHSA-vvjp-q62m-2vph CVE's with this release? Fix is to build with 1.21.5 or 1.21.12.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,35 @@
+## v2.11.0 / 2024-02-19
+
+### Note
+
+This release builds with Golang v1.21.7.
+
+ef8f1bfa: Dale Henries <dhenries@samaritan.org>: docs: add OpenSSF Scorecard to README
```
Comment (mrueg) [inline CHANGELOG.md:7]: Why did you change the format of the changelog here? Feel free to reuse https://github.com/kubernetes/kube-state-metrics/pull/2259

Comment (rexagod) [reply CHANGELOG.md:7]: Ah, I wanted to adhere to a format that can be generated by `git`, as that would be essential in [automating releases](https://github.com/kubernetes/kube-state-metrics/pull/2325/files#diff-226a12cd45d08a9b54de8d6c5bdfd74c461520624b2e2174806ef73d954f6c10R170).

Comment (mrueg) [reply CHANGELOG.md:7]: I would prefer, if we discuss these changes after the release and keep the current release process for now

Comment (rexagod) [reply CHANGELOG.md]: I've made the changes here and updated [#2325](https://github.com/kubernetes/kube-state-metrics/pull/2325/files#r1510879597) as well.

**Comment by mrueg** [comment]
Can you fix the markdown lint and rebase the release-2.11 branch on main first to reduce the delta to a minimum (that will help with merging it back)? https://github.com/kubernetes/kube-state-metrics/compare/release-2.11...main

**Comment by mrueg** [comment]
Also I would suggest to wait for tomorrow until go 1.21.8 is released, see: https://groups.google.com/g/golang-announce/c/smSYdsWaO4o

File: CHANGELOG.md (hunk 1 above)
Comment (mrueg) [reply CHANGELOG.md]: Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+* [ENHANCEMENT] Add restartPolicy to `kube_pod_init_container_info` metric by @changhyuni in <https://github.com/kubernetes/kube-state-metrics/pull/2240>
+* [FEATURE] Add `kube_node_status_addresses` metric by @stonith in <https://github.com/kubernetes/kube-state-metrics/pull/2252>
+* [FEATURE] Add namespace label to `endpointslice` metrics by @mrueg in <https://github.com/kubernetes/kube-state-metrics/pull/2266>
+* [FEATURE] Add opt-in `kube_persistentvolume_csi_attributes` metric by @machadovilaca in <https://github.com/kubernetes/kube-state-metrics/pull/2133>
+* [FEATURE] Add new metric `kube_pod_scheduler` metric by @adinhodovic in <https://github.com/kubernetes/kube-state-metrics/pull/2222>
+* [FEATURE] Support filtering annotations allow-list by `*` by @xonvanetta in <https://github.com/kubernetes/kube-state-metrics/pull/2234>
+* [ENHANCEMENT] Support scraping pod metrics that are still in scheduling status and have no assigned nodes by @mickeyzzc in <https://github.com/kubernetes/kube-state-metrics/pull/2217>
+* [FEATURE] Add backend resource info for `kube_ingress_path` by @rohitphatak in <https://github.com/kubernetes/kube-state-metrics/pull/2109>
+
```
Comment (mrueg) [inline CHANGELOG.md:17]: Can you add the https://github.com/kubernetes/kube-state-metrics/pull/2298 here as well?

**Comment by mrueg** [comment]
/lgtm

/hold 
for @rexagod to unhold.

**Comment by rexagod** [comment]
@dgrisonnet I believe this is good to go in now, but seeing as https://github.com/kubernetes/kube-state-metrics/pull/2270#issuecomment-1955093046 might be an `/lgtm` away, can we merge and include that here as well?

**Comment by rexagod** [comment]
/unhold
/lgtm

Will include #2270 in the next release, going forward with this one since this has been in the pipeline for a while.

**Pull request merged**